### ISSUE TITLE
CSV formats for event based risk outputs

### ISSUE THREAD

**Issue by raoanirudh** [gem/oq-engine#2150]
#### Stochastic event sets

The current CSV exporter for stochastic event sets provides very limited information regarding the ruptures. Instead, the stochastic event set parsers from [nrml_converters](https://github.com/GEMScienceTools/nrml_converters) should be used to generate the CSV outputs in a format similar to that listed below:

| event_id | event_set | tectonic_region | source_id | rupture_id | magnitude | centroid_lon | centroid_lat | centroid_depth | strike | dip | rake | boundary |
| :-: | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| 1 | 0003 | Subduction IntraSlab | Inslab_west_coast_4_1997_1997_0 | 251609-01 | 7.3 | -75.82 | 6.09 | 101.22 | 11.68 | 53.83 | -90 | MULTIPOLYGON(((-75.964360294 5.87265701971,-75.8674308742 6.33919340539,-75.6827673987 6.30122649939,-75.7798502417 5.83472306571,-75.964360294 5.87265701971))) |
| 2 | 0006 | Subduction IntraSlab | Inslab_west_coast_4_2123_2123_0 | 252497-01 | 7.3 | -75.74 | 6.18 | 103.34 | 11.72 | 60.48 | -90 | MULTIPOLYGON(((-75.8641816798 5.95834116722,-75.766942107 6.42481703028,-75.6127751264 6.39302933666,-75.7101446147 5.92658146426,-75.8641816798 5.95834116722))) |
| 3 | 0008 | Subduction IntraSlab | Inslab_west_coast_4_3246_3246_0 | 260998-01 | 7.8 | -75.27 | 5.96 | 159.35 | 11.53 | 80.82 | -90 | MULTIPOLYGON(((-75.3936883327 5.58000060929,-75.2339970802 6.35914892305,-75.1506570513 6.34224559907,-75.3104616445 5.56312127024,-75.3936883327 5.58000060929))) |
| 4 | 0013 | Subduction IntraSlab | Inslab_west_coast_4_3997_3997_1 | 266777-01 | 7.3 | -76.59 | 6.11 | 102.37 | 11.45 | 40.12 | -90 | MULTIPOLYGON(((-76.752728967 5.90030145191,-76.6577578255 6.36723670654,-76.4182802189 6.31902689747,-76.5134516092 5.85213367763,-76.752728967 5.90030145191))) |
| 5 | 0015 | Subduction IntraSlab | Inslab_west_coast_4_4996_4996_1 | 274480-01 | 7.2 | -76.04 | 6.01 | 109.33 | 11.44 | 49.51 | -90 | MULTIPOLYGON(((-76.1773516647 5.81523000143,-76.0916263929 6.23668628263,-75.9081132624 6.19973707744,-75.9939747144 5.77830938665,-76.1773516647 5.81523000143))) |
| 6 | 0018 | Active Shallow Crust | 10-198 | 87949-02 | 6.1 | -75.64 | 6.3 | 15 | 0 | 75 | 90 | MULTIPOLYGON(((-75.6487770861 6.25263075089,-75.6487790588 6.34132899536,-75.6256809412 6.34132899536,-75.6256829139 6.25263075089,-75.6487770861 6.25263075089))) |
| 7 | 0024 | Subduction IntraSlab | Inslab_west_coast_4_1496_1496_0 | 248132-01 | 7.6 | -76.11 | 6.01 | 92.89 | 11.42 | 44.5 | -90 | MULTIPOLYGON(((-76.3279602319 5.72376813013,-76.1991649444 6.35880955777,-75.8953715976 6.297786837,-76.024506756 5.66281658058,-76.3279602319 5.72376813013))) |
| 8 | 0024 | Subduction IntraSlab | Inslab_west_coast_4_1622_1622_0 | 249017-01 | 7.3 | -76.03 | 6.09 | 94.06 | 11.47 | 48.95 | -90 | MULTIPOLYGON(((-76.1754520921 5.88154679751,-76.0802395766 6.34843287545,-75.8746086707 6.30693151402,-75.9699926415 5.8400815261,-76.1754520921 5.88154679751))) |
| 9 | 0029 | Subduction IntraSlab | Inslab_west_coast_4_4749_4749_1 | 272222-01 | 7 | -76.13 | 6.27 | 106.41 | 11.5 | 41.62 | -90 | MULTIPOLYGON(((-76.2507818123 6.11398945158,-76.1805452642 6.45727878651,-76.0083665144 6.42244271999,-76.0787116429 6.0791763027,-76.2507818123 6.11398945158))) |
| 10 | 0031 | Subduction IntraSlab | Inslab_west_coast_4_6495_6495_1 | 285667-01 | 7.5 | -75.36 | 5.87 | 156.3 | 191.6 | 89.23 | -90 | MULTIPOLYGON(((-75.2957087375 6.15428604022,-75.4138613387 5.58148286361,-75.4190064864 5.58253300812,-75.300858952 6.15533726438,-75.2957087375 6.15428604022))) |
| 11 | 0033 | Active Shallow Crust | 10-207 | 88386-01 | 5.9 | -75.64 | 6.21 | 5 | 0 | 75 | 90 | MULTIPOLYGON(((-75.6469430962 6.17164955635,-75.6469443341 6.24243028439,-75.6285156659 6.24243028439,-75.6285169038 6.17164955635,-75.6469430962 6.17164955635))) |
| 12 | 0036 | Active Shallow Crust | 10-216 | 88780-01 | 5.5 | -75.64 | 6.12 | 15 | 0 | 75 | 90 | MULTIPOLYGON(((-75.6440860106 6.09457354283,-75.6440865052 6.13964639353,-75.6323534948 6.13964639353,-75.6323539894 6.09457354283,-75.6440860106 6.09457354283))) |
| 13 | 0039 | Subduction IntraSlab | Inslab_west_coast_4_1736_1736_0 | 249620-01 | 7.6 | -76.13 | 5.12 | 103.1 | 9.65 | 60.27 | -90 | MULTIPOLYGON(((-76.2909420014 4.82061890949,-76.1819715175 5.45929199904,-75.9698820245 5.42335860559,-76.0790581832 4.78472140042,-76.2909420014 4.82061890949))) |
| 14 | 0039 | Subduction IntraSlab | Inslab_west_coast_4_2123_2123_0 | 252497-01 | 7.3 | -75.74 | 6.18 | 103.34 | 11.72 | 60.48 | -90 | MULTIPOLYGON(((-75.8641816798 5.95834116722,-75.766942107 6.42481703028,-75.6127751264 6.39302933666,-75.7101446147 5.92658146426,-75.8641816798 5.95834116722))) |
| 15 | 0045 | Subduction IntraSlab | Inslab_west_coast_4_3997_3997_1 | 266777-01 | 7.3 | -76.59 | 6.11 | 102.37 | 11.45 | 40.12 | -90 | MULTIPOLYGON(((-76.752728967 5.90030145191,-76.6577578255 6.36723670654,-76.4182802189 6.31902689747,-76.5134516092 5.85213367763,-76.752728967 5.90030145191))) |
| 16 | 0045 | Active Shallow Crust | 10-199 | 87987-01 | 5.9 | -75.55 | 6.3 | 5 | 0 | 75 | 90 | MULTIPOLYGON(((-75.5559646717 6.26158955519,-75.5559659279 6.33237028323,-75.5375340721 6.33237028323,-75.5375353283 6.26158955519,-75.5559646717 6.26158955519))) |
| 17 | 0050 | Active Shallow Crust | 10-207 | 88384-01 | 5.7 | -75.64 | 6.21 | 15 | 0 | 75 | 90 | MULTIPOLYGON(((-75.6450821012 6.17879862678,-75.6450828895 6.23528127181,-75.6303771105 6.23528127181,-75.6303778988 6.17879862678,-75.6450821012 6.17879862678))) |
| 18 | 0054 | Subduction IntraSlab | Inslab_west_coast_4_2622_2622_0 | 256743-01 | 7.1 | -75.52 | 6.08 | 120.29 | 11.93 | 70.82 | -90 | MULTIPOLYGON(((-75.5998234889 5.89452745998,-75.5191537511 6.27426277885,-75.4354812284 6.25668013714,-75.5162073362 5.87695721716,-75.5998234889 5.89452745998))) |
| 19 | 0059 | Subduction IntraSlab | Inslab_west_coast_4_369_369_0 | 264211-01 | 7.5 | -76.77 | 5.87 | 84.89 | 11.3 | 41.67 | -90 | MULTIPOLYGON(((-76.9748672111 5.61436491923,-76.8597884836 6.18778774126,-76.572606364 6.13068322008,-76.6879687502 5.55731914816,-76.9748672111 5.61436491923))) |
| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |

**Comment by raoanirudh** [comment]
#### Aggregate event loss tables

Currently, the aggregate loss tables for different loss categories are provided in separate files. These could be combined into a single CSV file for each realization. Also, instead of the rupture tags, some useful information about the ruptures, such as the magnitude and hypocenter location, should be provided directly in the event loss table CSV files so that the users can avoid having to merge multiple files in order to get the relevant information they need.

| event_id | event_set | tectonic_region      | magnitude | centroid_lon | centroid_lat | centroid_depth | contents | nonstructural | occupants | structural | structural_ins | … |
|----------|-----------|----------------------|-----------|--------------|--------------|----------------|----------|---------------|-----------|------------|----------------|---|
| 1        | 0003      | Subduction IntraSlab | 7.3       | -75.82       | 6.09         | 101.22         | 4.21E+08 | 8.44E+08      | 2.49E+06  | 1.02E+09   | 6.11E+08       | … |
| 2        | 0006      | Subduction IntraSlab | 7.3       | -75.74       | 6.18         | 103.34         | 5.15E+09 | 1.03E+10      | 3.04E+07  | 1.25E+10   | 7.48E+09       | … |
| 3        | 0008      | Subduction IntraSlab | 7.8       | -75.27       | 5.96         | 159.35         | 4.65E+09 | 9.34E+09      | 2.75E+07  | 1.13E+10   | 6.76E+09       | … |
| 4        | 0013      | Subduction IntraSlab | 7.3       | -76.59       | 6.11         | 102.37         | 6.86E+09 | 1.38E+10      | 4.06E+07  | 1.66E+10   | 9.97E+09       | … |
| 5        | 0015      | Subduction IntraSlab | 7.2       | -76.04       | 6.01         | 109.33         | 4.46E+09 | 8.95E+09      | 2.64E+07  | 1.08E+10   | 6.48E+09       | … |
| 6        | 0018      | Active Shallow Crust | 6.1       | -75.64       | 6.30         | 15.00          | 4.17E+08 | 8.37E+08      | 2.47E+06  | 1.01E+09   | 6.06E+08       | … |
| 7        | 0024      | Subduction IntraSlab | 7.6       | -76.11       | 6.01         | 92.89          | 1.88E+08 | 3.77E+08      | 1.11E+06  | 4.55E+08   | 2.73E+08       | … |
| 8        | 0024      | Subduction IntraSlab | 7.3       | -76.03       | 6.09         | 94.06          | 4.37E+09 | 8.77E+09      | 2.58E+07  | 1.06E+10   | 6.35E+09       | … |
| 9        | 0029      | Subduction IntraSlab | 7.0       | -76.13       | 6.27         | 106.41         | 4.80E+09 | 9.63E+09      | 2.84E+07  | 1.16E+10   | 6.97E+09       | … |
| 10       | 0031      | Subduction IntraSlab | 7.5       | -75.36       | 5.87         | 156.30         | 4.16E+08 | 8.35E+08      | 2.46E+06  | 1.01E+09   | 6.04E+08       | … |
| 11       | 0033      | Active Shallow Crust | 5.9       | -75.64       | 6.21         | 5.00           | 4.17E+08 | 8.38E+08      | 2.47E+06  | 1.01E+09   | 6.06E+08       | … |
| 12       | 0036      | Active Shallow Crust | 5.5       | -75.64       | 6.12         | 15.00          | 4.65E+09 | 9.34E+09      | 2.75E+07  | 1.13E+10   | 6.76E+09       | … |
| 13       | 0039      | Subduction IntraSlab | 7.6       | -76.13       | 5.12         | 103.10         | 4.21E+08 | 8.46E+08      | 2.49E+06  | 1.02E+09   | 6.12E+08       | … |
| 14       | 0039      | Subduction IntraSlab | 7.3       | -75.74       | 6.18         | 103.34         | 4.14E+09 | 8.31E+09      | 2.45E+07  | 1.00E+10   | 6.02E+09       | … |
| 15       | 0045      | Subduction IntraSlab | 7.3       | -76.59       | 6.11         | 102.37         | 1.80E+08 | 3.62E+08      | 1.06E+06  | 4.36E+08   | 2.62E+08       | … |
| 16       | 0045      | Active Shallow Crust | 5.9       | -75.55       | 6.30         | 5.00           | 4.18E+08 | 8.38E+08      | 2.47E+06  | 1.01E+09   | 6.06E+08       | … |
| 17       | 0050      | Active Shallow Crust | 5.7       | -75.64       | 6.21         | 15.00          | 4.21E+09 | 8.46E+09      | 2.49E+07  | 1.02E+10   | 6.12E+09       | … |
| 18       | 0054      | Subduction IntraSlab | 7.1       | -75.52       | 6.08         | 120.29         | 4.21E+08 | 8.45E+08      | 2.49E+06  | 1.02E+09   | 6.11E+08       | … |
| 19       | 0059      | Subduction IntraSlab | 7.5       | -76.77       | 5.87         | 84.89          | 4.24E+08 | 8.51E+08      | 2.50E+06  | 1.03E+09   | 6.15E+08       | … |
| ⋮        | ⋮         | ⋮                    | ⋮         | ⋮            | ⋮            | ⋮              | ⋮        | ⋮             | ⋮         | ⋮          | ⋮              | ⋱ |

**Comment by raoanirudh** [comment]
The event set exporter implemented in #2336 resolves the first part of this issue. And after #2352, the event loss tables will need to include the event year as well. So I propose closing this issue and opening a separate issue for the CSV exporter for the event loss tables.
